app/frontend/src/features/profile/HostingPreferenceForm.tsx

```diff
@@ -1,10 +1,231 @@
+            className={classes.formControl}
+            defaultValue={!!user.acceptsPets?.value}
+            label="Pets OK"
+            name="acceptsPets"
+            register={register}
+          />
+          <HostingPreferenceCheckbox
+            className={classes.formControl}
+            defaultValue={!!user.lastMinute?.value}
+            label="Last minute requests OK"
+            name="lastMinute"
+            register={register}
+          />
+          <HostingPreferenceCheckbox
+            className={classes.formControl}
+            defaultValue={!!user.wheelchairAccessible?.value}
+            label="Wheelchair accessible"
+            name="wheelchairAccessible"
+            register={register}
+          />
+          <Controller
+            control={control}
+            defaultValue={user.maxGuests?.value ?? null}
+            name="maxGuests"
+            render={({ onChange }) => (
+              <Autocomplete
+                disableClearable={false}
+                defaultValue={user.maxGuests?.value}
+                forcePopupIcon
+                freeSolo
+                getOptionLabel={(option) => option.toString()}
+                options={[0, 1, 2, 3, 4, 5]}
+                onChange={(e, value) => onChange(value)}
+                multiple={false}
+                renderInput={(params) => (
+                  <ProfileTextInput
+                    {...params}
+                    error={!!errors?.maxGuests?.message}
+                    helperText={errors?.maxGuests?.message}
+                    label="Max. number of guests"
+                    name="maxGuests"
+                    onChange={(e) => onChange(Number(e.target.value))}
+                  />
+                )}
+              />
+            )}
+            rules={{
+              validate: (value) =>
+                isNaN(value) ? "Invalid number provided" : true,
+            }}
+          />
+          <ProfileTextInput
+            label="Description of your place"
+            name="area"
+            defaultValue={user.area?.value ?? ""}
+            inputRef={register}
+            rowsMax={5}
+            multiline
+          />
+          <ProfileTextInput
+            label="House rules"
+            name="houseRules"
+            defaultValue={user.houseRules?.value ?? ""}
+            inputRef={register}
+            rowsMax={5}
+            multiline
+          />
+          <Controller
+            control={control}
+            defaultValue={user.smokingAllowed}
+            name="smokingAllowed"
+            render={({ onChange }) => (
+              <Autocomplete
+                disableClearable
+                defaultValue={user.smokingAllowed}
+                forcePopupIcon
+                freeSolo={false}
+                getOptionLabel={(option) => smokingLocationLabels[option]}
+                multiple={false}
+                options={[
+                  SmokingLocation.SMOKING_LOCATION_YES,
```

Comment:
I think UNSPECIFIED should be an option, for the user to "unset" it (and maybe make the label a blank string). In this case, it wouldn't be shown on the profile.

Reply:
Hmm not sure if this is a bug in the generated client again, but it's not working when the option has a value of 0, the value `UNSPECIFIED` has...

Reply:
Yeah I double checked and the value 0 (for `UNSPECIFIED`) is captured correctly in the form, and it's the only value that doesn't work. I think it's better to leave this out as this would introduce a bug.

Reply:
It's been a while since I looked at that backend code, but I think it should never be set as zero, zero is like "null". Maybe zero is meant to be unknown and 1 should be unspecified? Or maybe the labels are just not good

Reply:
I imagine we'll have to update the proto file for the value to start at 1, since it's generated code?

Reply:
From the proto:
```
enum SmokingLocation {
  SMOKING_LOCATION_UNSPECIFIED = 0;
  SMOKING_LOCATION_UNKNOWN = 1;
  SMOKING_LOCATION_YES = 2;
  SMOKING_LOCATION_WINDOW = 3;
  SMOKING_LOCATION_OUTSIDE = 4;
  SMOKING_LOCATION_NO = 5;
}
```
We should probably swap `UNKNOWN` and `UNSPECIFIED` around then

Reply:
Looking at the backend tests, it looks like this is designed to be cleared by setting the value to `SMOKING_LOCATION_UNKNOWN`.

Rather than adding another option with a blank string (bad for accessibility when I thought about it more), I've allowed the field to be clearable instead and when the user clears, it will set the value to `SMOKING_LOCATION_UNKNOWN`.

Reply:
Yes exactly. It's my fault, I made those enums and I should have done them the other way around...